apple-dmg/src/blkx.rs

```diff
@@ -0,0 +1,216 @@
+use crate::koly::UdifChecksum;
+use anyhow::Result;
```

Comment:
I have a preference for avoiding `anyhow` for _library_ crates. But we could potentially migrate to something like `thiserror` after this merges.

Reply:
well, I think migrating to something like `thiserror` instead of just returning strings everywhere is a good idea.
```
#[derive(Debug, Error)]
#[error = "custom message"]
pub struct ACustomError;
```
but I still prefer the result type to be `anyhow::Result` so that backtrace capture works automatically. it also massively simplifies types when you have traits, it's heap allocated which means that the stack which ends up in a cpu cache doesn't get trashed on the happy path. in addition any error handling for an os error like wouldblock or interrupted should be handled as close to the source as possible, things that are expected should be handled with options/enums in the Ok path. this leaves the error path with transient/fatal errors that are logged and maybe retried.